Backported from GNOME #356435; fixes crash triggerred by missing NULL
terminator in name list.

2006-12-10  Luca Cavalli  <lcavalli@cvs.gnome.org>

       * font-method.c: (ensure_font_list): make sure the name list array
       is NULL terminated. Fixes #356435.

diff -Nur control-center-2.14.2/vfs-methods/fontilus/font-method.c control-center-2.14.2.new/vfs-methods/fontilus/font-method.c
--- control-center-2.14.2/vfs-methods/fontilus/font-method.c	2005-11-14 16:29:27.000000000 +0100
+++ control-center-2.14.2.new/vfs-methods/fontilus/font-method.c	2007-02-26 10:58:25.000000000 +0100
@@ -171,8 +171,8 @@
 	goto end;
     }
 
-    /* set up name list and hash */
-    font_names = g_new(gchar *, font_list->nfont);
+    /* set up name list and hash (name list array must be NULL terminated) */
+    font_names = g_new0(gchar *, font_list->nfont + 1);
     font_hash = g_hash_table_new(g_str_hash, g_str_equal);
     for (i = 0; i < font_list->nfont; i++) {
 	font_names[i] = get_pango_name(font_list->fonts[i]);
